### DISCUSSION TITLE
Reason why JsonContent (From CoreFx) method TryComputeLength is False

### DISCUSSION THREAD

**Discussion by moisoiu** [dotnet/runtime#43883] [Q&A] [answered]
Hello,

I'm having a hard time understanding why `TryComputeLength `from `JsonContent ` is on `false` and `length `is 0.

Scenario.
The application that I'm working on uses `PostAsJsonAsync` from `System.Net.Http.Json.HttpClientJsonExtensions` and I'm using it to call different backends.

If I run locally, I have no issue, I receive an answer from backend on which I make the request. But as soon as I deployed in Azure App Service, it returns me bad request. And I don't understand why

The reason why I'm asking is because If I create a mockup `JsonContent ` and change the `TryComputeLength `from `false` to `true` and set the `length `manually it **works**

The only solution that I found is to replace `JsonContent ` with `ByteArrayContent` 

Also on the `Bad Request `it returns me a header `Transfer-Encoding: chunked `

So anyone any idea, because I don't think is a bug in the JsonContent but in the same time I don't understand why it works locally but not on Azure App Service

**Comment by huoyaoyuan** [comment, marked as answer]
`TryComputeLength` means if the length can be given **immediately**. Serializing arbitrary content may consume time that's sensitive for control flow.

You can use `LoadIntoBufferAsync()` to do the serialization. It will store the result internally, then the length can be get immediately. See my code [here](https://github.com/huoyaoyuan/ReunionGet/blob/2ccf2cde5112483fc86bf2d49c4d9c368a6b5761/Source/Portable/ReunionGet.Aria2Rpc/Aria2Connection.cs#L76-L78).

It also surprises me when I was debugging my code, because inspecting the json content in debugger will make the code execute successfully.

Many code supposes chunked encoding is supported everywhere, because it's a part of HTTP spec. So does `HttpClient`.

**Reply by moisoiu** [reply]
You're right, after further investigation we found the cause. It seems like this was a Firewall configuration where for some reason is not allowing this type of requests. 

Thanks for the tips with `LoadIntoBufferAsync` !

**Reply by stuartclode** [reply]
Great catch. Thanks for figuring this out.
